### ISSUE TITLE
BTM masks original SQLException

### ISSUE THREAD

**Issue by torbenriis** [bitronix/btm#60]
We have experienced that BTM masks some SQLException’s if database errors occurs.
Further investigations of the source code has shown that is only occurs in LrcXAResource, and in all occurrences, every time around auto commit handling.

Are there an explicit reason why the original cause is masked away? The cause could contain any reason from internal DB errors to lost connection.

**Snippet**
`        try {`
`               connection.setAutoCommit(true);`
`        } catch (SQLException ex) {`
`            throw new BitronixXAException("cannot reset autocommit on non-XA connection", XAException.XAER_RMERR);`
`        }`

**Comment by lorban** [comment]
No, there is no reason: this seems to be an overlook as `BitronixXAException` accepts a `Throwable` as a 3rd argument.

You're welcome to submit a pull request with a fix!
